wrangler-core/src/test/java/io/cdap/functions/JsonFunctionsTest.java

```diff
@@ -0,0 +1,94 @@
+ *  the License.
+ */
+
+package io.cdap.functions;
+
+import io.cdap.wrangler.TestingRig;
+import io.cdap.wrangler.api.Row;
+import io.cdap.wrangler.utils.JsonTestData;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Tests for all the Json functions.
+ */
+public class JsonFunctionsTest {
+
+  private static final String JSON_SELECTION_EG1 = "{\n" +
+                                                     "  \"list\" : [\n" +
+                                                     "    {\n" +
+                                                     "      \"set\" : [\n" +
+                                                     "        { \"a1\" : \"b1\" },\n" +
+                                                     "        { \"x1\" : \"y1\" }\n" +
+                                                     "      ],\n" +
+                                                     "      \"map\" : \"X1\",\n" +
+                                                     "      \"collection\" : \"Y1\"\n" +
+                                                     "    },\n" +
+                                                     "    {\n" +
+                                                     "      \"set\" : [\n" +
+                                                     "        { \"a2\" : \"b2\" },\n" +
+                                                     "        { \"x2\" : \"y2\" }\n" +
+                                                     "      ],\n" +
+                                                     "      \"map\" : \"X2\",\n" +
+                                                     "      \"collection\" : \"Y2\"      \n" +
+                                                     "    },\n" +
+                                                     "    {\n" +
+                                                     "      \"set\" : [\n" +
+                                                     "        { \"a3\" : \"b3\" },\n" +
+                                                     "        { \"x3\" : \"y3\" }\n" +
+                                                     "      ],\n" +
+                                                     "      \"map\" : \"X3\",\n" +
+                                                     "      \"collection\" : \"Y3\"      \n" +
+                                                     "    }\n" +
+                                                     "  ]\n" +
+                                                     "}";
+
+  @Test
+  public void testJsonSelect() throws Exception {
+    String[] directives = new String[]{
+      "set-column mayo json:Parse(body)",
+      "drop boody",
+      "set-column entries json:Select(mayo, '$.list[*].set.*')"
+    };
+    List<Row> rows = Arrays.asList(
+      new Row("body", JSON_SELECTION_EG1)
+    );
+    rows = TestingRig.execute(directives, rows);
+    Assert.assertTrue(rows.size() == 1);
```

Comment:
The directives could be doing all sorts of wrong things and this test would still pass. It should be checking the output.